src/Umbraco.Core/UmbracoApplicationBase.cs

```diff
@@ -59,6 +62,22 @@ internal void StartApplication(object sender, EventArgs e)
         }
 
         /// <summary>
+        /// Called when an assembly can't be resolved. In here we can do magic with the assembly name and try loading another.
+        /// This is used for loading a signed assembly of AutoMapper (v. 3.1+) without having to recompile old code.
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="args"></param>
+        /// <returns></returns>
+        private Assembly CurrentDomainOnAssemblyResolve(object sender, ResolveEventArgs args)
+        {
+            // ensure the assembly is indeed AutoMapper and that the PublicKeyToken is null before trying to Load again
+            // do NOT just replace this with 'return Assembly', as it will cause an infinite loop -> stackoverflow
+            if (args.Name.StartsWith("AutoMapper") && args.Name.EndsWith("PublicKeyToken=null"))

```

Comment:
Yep - only reason why I changed away from the regex to the simple string match was because I didn't want to load up a regex (for performance) but then I later realized this will of course only run once and in very limited scenarios.... so no reason at all trying to optimize it :)